src/context/files.tsx

```diff
@@ -0,0 +1,152 @@
+/* eslint-disable no-console */
+import React, { createContext, useState, useCallback, useContext } from 'react'
+import filesize from 'filesize'
+
+import { api, Endpoints } from '@/services/api/index'
+import { useAuth } from './auth'
+
+export interface IFile {
+  userId: number
+  id?: number
+  filename: string
+  mimeType: string
+  size: string
+  uploaded: boolean
+  error: boolean
+  progress?: number
+  file: File
+}
+
+interface IFileContextData {
+  uploadedFiles: IFile[]
+  deleteFile(id: number): void
+  handleUpload(file: any, userId: number): void
+  loadUserFiles(userId: number): void
+  processUpload(file: IFile): void
+}
+
+const FileContext = createContext<IFileContextData>({} as IFileContextData)
+
+const FileProvider: React.FC = ({ children }) => {
+  const [uploadedFiles, setUploadedFiles] = useState<IFile[]>([])
+  const { user } = useAuth()
+
+  const updateFile = useCallback((id, data) => {
+    setUploadedFiles((state) => state.map((file) => (file.id === id ? { ...file, ...data } : file)))
+  }, [])
+
+  const loadUserFiles = useCallback(
+    (userId: number) => {
+      api
+        .get<IFile[]>(Endpoints.GET_FILES, {
+          params: {
+            user_id: userId,
+          },
+        })
+        .then((response) => {
```

Comment:
I'm curious is it possible to use `await` here?

Reply:
It would be, yes! But as I'm using mostly useCallback for this exact type of calls, I didn't. The promises works finely enough, tho.

Why, you ask? I'd say to ease my way out in this deliver - that's what I thought at that moment, lol. useCallback with promises return just simplified it for me in there.